apps/omg/lib/omg/state/transaction/fee.ex

```diff
@@ -0,0 +1,114 @@
+
+  defstruct [:outputs]
+
+  @type t() :: %__MODULE__{outputs: [Output.FungibleMoreVPToken.t() | Output.UniquenessEnforcer.t()]}
+
+  @doc """
+  Creates new fee claiming transaction
+  """
+  @spec new(
+          blknum :: non_neg_integer(),
+          {Crypto.address_t(), Transaction.Payment.currency(), pos_integer}
+        ) :: t()
+  def new(blknum, {owner, currency, amount}) do
+    %__MODULE__{
+      outputs: [
+        %Output.UniquenessEnforcer{blknum: blknum, type_marker: @uniqueness_output_type_marker},
+        %Output.FungibleMoreVPToken{
+          owner: owner,
+          currency: currency,
+          amount: amount,
+          type_marker: @fee_output_type_marker
+        }
+      ]
+    }
+  end
+
+  @doc """
+  Transaform the structure of RLP items after a successful RLP decode of a raw transaction, into a structure instance
+  """
+  def reconstruct([outputs_rlp]) do
+    with {:ok, outputs} <- reconstruct_outputs(outputs_rlp),
+         do: {:ok, %__MODULE__{outputs: outputs}}
+  end
+
+  def reconstruct(_), do: {:error, :malformed_transaction}
+
+  defp reconstruct_outputs(outputs_rlp) do
+    with {:ok, outputs} <- parse_outputs(outputs_rlp),
+         do: {:ok, outputs}
+  end
+
+  defp parse_outputs(outputs_rlp) do
+    outputs = Enum.map(outputs_rlp, &Output.dispatching_reconstruct/1)
+
+    with nil <- Enum.find(outputs, &match?({:error, _}, &1)),
+         true <- only_allowed_output_types?(outputs) || {:error, :tx_cannot_create_output_type},
+         do: {:ok, outputs}
+  rescue
+    _ -> {:error, :malformed_outputs}
+  end
+
+  defp only_allowed_output_types?([%Output.UniquenessEnforcer{}, %Output.FungibleMoreVPToken{}]), do: true
+  defp only_allowed_output_types?(_), do: false
+end
+
+defimpl OMG.State.Transaction.Protocol, for: OMG.State.Transaction.Fee do
+  alias OMG.InputPointer
+  alias OMG.Output
+  alias OMG.State.Transaction
+
+  @fee_tx_type_marker <<0xFE, 0xE0>>
+
+  @doc """
+  Turns a structure instance into a structure of RLP items, ready to be RLP encoded, for a raw transaction
+  """
+  @spec get_data_for_rlp(Transaction.Fee.t()) :: list(any())
+  def get_data_for_rlp(%Transaction.Fee{outputs: outputs}),
+    do: [
+      @fee_tx_type_marker,
+      Enum.map(outputs, &OMG.Output.Protocol.get_data_for_rlp/1)
+    ]
+
+  @spec get_outputs(Transaction.Fee.t()) :: list(Output.Protocol.t())
+  def get_outputs(%Transaction.Fee{outputs: outputs}), do: outputs
+
+  @spec get_inputs(Transaction.Fee.t()) :: list(InputPointer.Protocol.t())
+  def get_inputs(%Transaction.Fee{}), do: []
+
+  @doc """
+  Fee claiming transaction is not used to transfer funds
+  """
+  @spec valid?(Transaction.Fee.t(), Transaction.Signed.t()) :: {:error, atom()}
+  def valid?(%Transaction.Fee{}, _signed_tx), do: {:error, :transaction_not_transfer_funds}
+
+  @doc """
+  Fee claiming transaction is not used to transfer funds
+  """
+  @spec can_apply?(Transaction.Fee.t(), list(Output.Protocol.t())) :: {:error, atom()}
```

Comment:
same here - `Fee` can't be ever applied?

Reply:
Task was splited, I'll refine the proper handling of this doing Watcher part. WDYT?